core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -289,7 +289,8 @@ case class SparkListenerApplicationStart(
     driverAttributes: Option[Map[String, String]] = None) extends SparkListenerEvent
 
 @DeveloperApi
-case class SparkListenerApplicationEnd(time: Long) extends SparkListenerEvent
+case class SparkListenerApplicationEnd(time: Long, exitCode: Option[Int] = None)
```

Comment:
we'll have to define another constructor instead of using default value which breaks binary compat (and +source compat if you use Java)